SIM/RetrievalManagement.md

```diff
@@ -0,0 +1,233 @@
+* RPC: This module is a Client side of the task manager. Resource
+  Manager make a procedural call to RPC methods(i.e sync_storage_resource()). Later, it cast the task to RabbitMq message bus.
+* Manager: It is tasks consumer part of the task manager. It receive the
+  tasks from RabbitMq and make a procedural call to appropriate methods based on received input tasks.
+* Tasks: It contains the implementation code for retrieving the
+  resources information and updating the DB. To get the latest information from storage backends, it calls driver manager methods, and later updates the DB with collected information.
+
+
+##### Delfin Driver manager:
+* This layer contains driver’s code. Drivers are lightweight processes
+  that configure and collect data from various sources including storage, switch , host etc.
+* Each driver’s connected to a single storage backend and performs
+  specific tasks(storage, volumes, pools collection) on that backend. Driver manager’s responsibility is to take input from the task manager and select specific drivers based on input parameters. Once the task is performed, return the response with collected information.
+* Task manager to driver manager is a procedural call. And from driver
+  manager to delfin south bound drivers are REST, CLI, SNMP, SSH, SMI etc.
+
+
+#### Interface Model
+
+##### External Driver Interfaces
+
+The interface for storage/volume/pool collections:
+
+1. driver_api.get_storage(context, storage_id)
+
+2. driver_api.list_pools(context, storage_id)
+
+3. driver_api.list_volumes(context, storage_id)
+
+
+##### Databases Interfaces:
+###### DB Storage Interface
+1. db.storage_get(context, storage_id)
+
+2. db.storage_update(context, storage_id, storage)
+
+3. db.storage_delete(context, storage_id)
+
+###### DB Pool Interface
+4. db.storage_pool_get_all(context, filters={"storage_id": storage_id})
+
+5. db.storage_pools_delete(context, delete_id_list)
+
+6. db.storage_pools_update(context, update_list)
+
+7. db.storage_pools_create(context, add_list)
+
+###### DB Volume Interface
+8. db.storage_volume_get_all(context, filters={"storage_id": storage_id})
+
+9. db.storage_volumes_delete(context, delete_id_list)
+
+10. db.storage_volumes_update(context, update_list)
+
+11. db.storage_volumes_create(context, add_list)
+
+###### DB AccessInfo Interface
+12. db.access_info_delete(context, storage_id)
+
+###### DB Alertsource Interfaces
+13. db.alert_source_delete(context, storage_id)
+
+
+### Data View
+
+#### Data Model
+Data Model can be referred from [here](https://github.com/sodafoundation/design-specs/blob/master/specs/SIM/SODA_InfrastructureManagerDesign.md#data-model)
+
+
+### Development and Deployment Context
+
+#### Code
+The module code can be found [here](https://github.com/sodafoundation/delfin/tree/master/delfin/task_manager)
+
+### Execution View
+
+NA
+
+
+## Sequence Diagrams
+
+Below are sequence diagrams of the resource management module.
```

Comment:
In the sequence diagram, some of the arrows may need both end pointers..Like for step 2, there is response too, right?
Also, the purpose of take and release lock is not clearly mentioned above and it's simply put here..Do we need the explanation for the locking?
Question:
Consider a case of syncVolume(): One request came and went past step 6 and has reached at 11. Now another request comes and take a lock now. At step 12, it will try to take the lock and doesn't get it as another task has taken the lock. What will be the behavior now? The data has been sent to exporter but not yet synced. 
At step it will retry? Can there be starvation?